src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -79,6 +79,36 @@ def customDID(pname):
     return pname
 
 
+def updateTransitionRecord(doc, userDN, logger):
+    """
+    Update transition record based on the following logic:
+    - check previous fraction value and add new transition record if it is
+    - differ from current one or does not exist
+      - first use-case: differ from container fraction of record
+      - second use-case: differ from last transition container fraction record
+    - https://gist.github.com/amaltaro/b4f9bafc0b58c10092a0735c635538b5
+
+    :param doc: MSPileup record
+    :param userDN: user DN (string)
+    :param logger: logger to use
+    :return: nothing (document transition is updated in place)
+    """
+    fraction = doc['containerFraction']
+    transition = doc.get('transition', [])
+    prevTranRecord = transition[-1]
```

Comment:
If the transition field is empty, this line will hit an IndexError exception. Given that we discussed that no document should exist without a transition record, perhaps you can merge these two lines into:
```
prevTranRecord = doc['transition'][-1]
```

Reply:
ok